nodebuilder/state/core.go

```diff
@@ -20,7 +20,17 @@ func coreAccessor(
 	sync *sync.Syncer[*header.ExtendedHeader],
 	fraudServ libfraud.Service[*header.ExtendedHeader],
 ) (*state.CoreAccessor, Module, *modfraud.ServiceBreaker[*state.CoreAccessor, *header.ExtendedHeader]) {
-	ca := state.NewCoreAccessor(signer, sync, corecfg.IP, corecfg.RPCPort, corecfg.GRPCPort)
+	ca := state.NewCoreAccessor(
+		signer,
+		sync,
+		corecfg.RPC.Scheme,
+		corecfg.RPCHost(),
+		corecfg.RPC.Port,
+		corecfg.GRPC.Scheme,
+		corecfg.GRPCHost(),
+		corecfg.GRPC.Port,
+		corecfg.GRPC.Cert,
```

Comment:
not a super big fan of this but copying what i _think_ the existing patterns are, which i believe is to deliberately passing in longer lists of values to keep these nodebuilder module decoupled from one another. I wonder if it's preferable to use the core/config.HostConfig i introduced and passing one each of those in here, reducing it to just two args